File: MonRoadtrip-back/src/main/java/monRoadtrip/model/Compte.java
package monRoadtrip.model;

import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.SequenceGenerator;
import javax.persistence.Version;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@SequenceGenerator(name = "seqCompteJPA",sequenceName = "seqCompte")
public abstract class Compte {

	@Id
	//@GeneratedValue(strategy = GenerationType.SEQUENCE,generator = "seqCompteJPA")
	@GeneratedValue(strategy = GenerationType.TABLE)
    protected Integer id;

    protected String nom;
    protected String prenom;
	protected String mail;
    protected String password;
    @Column(name="date_naissance")
    protected LocalDate dateNaissance;
    
    @Version
    protected int version;
    
    public Compte() {
	}
    
	public Compte(Integer id,String nom, String prenom, String mail, String password, LocalDate dateNaissance) {
		this.nom = nom;
		this.prenom = prenom;
		this.mail = mail;
		this.password = password;
		this.id=id;
		this.dateNaissance = dateNaissance;
	}
	
	public Compte(String nom, String prenom, String mail, String password, LocalDate dateNaissance) {
		this.nom = nom;
		this.prenom = prenom;
		this.mail = mail;
		this.password = password;
		this.dateNaissance = dateNaissance;
	}
			
	public String getNom() {
		return nom;
	}
	public void setNom(String nom) {
		this.nom = nom;
	}
	public String getPrenom() {
		return prenom;
	}
	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}
	public String getMail() {
		return mail;
	}
	public void setMail(String mail) {
		this.mail = mail;
	}
	public String getPassword() {
		return password;
	}
	public void setPassword(String password) {
		this.password = password;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public LocalDate getDateNaissance() {
		return dateNaissance;
	}
	public void setDateNaissance(LocalDate dateNaissance) {
		this.dateNaissance = dateNaissance;
	}
	

    

}

File: monRoadTripAngular/src/app/model/compte.ts
import { Adresse } from './adresse';
export class Compte {
  constructor(
    private _id?: number | undefined,
    private _dateNaissance?: Date | undefined,
    private _mail?: string | undefined,
    private _nom?: string | undefined,
    private _password?: string | undefined,
    private _prenom?: string | undefined,
    private _adresse?: Adresse | undefined,
    private _solde?: number | undefined,
    private _typePaiement?: string | undefined, //a passer en enum,
    private _type?: string | undefined
  ) {}

  public get id(): number | undefined {
    return this._id;
  }

  public get dateNaissance(): Date | undefined {
    return this._dateNaissance;
  }

  public get mail(): string | undefined {
    return this._mail;
  }

  public get nom(): string | undefined {
    return this._nom;
  }

  public get password(): string | undefined {
    return this._password;
  }

  public get prenom(): string | undefined {
    return this._prenom;
  }

  public get adresse(): Adresse | undefined {
    return this._adresse;
  }

  public get solde(): number | undefined {
    return this._solde;
  }

  public get typePaiement(): string | undefined {
    return this._typePaiement;
  }

  public get type(): string | undefined {
    return this._type;
  }

  public set id(value: number | undefined) {
    this._id = value;
  }
  public set mail(value: string | undefined) {
    this._mail = value;
  }
  public set password(value: string | undefined) {
    this._password = value;
  }
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }
  public set nom(value: string | undefined) {
    this._nom = value;
  }
  public set adresse(value: Adresse | undefined) {
    this._adresse = value;
  }
  public set solde(value: number | undefined) {
    this._solde = value;
  }
  public set typePaiement(value: string | undefined) {
    this._typePaiement = value;
  }

  public set dateNaissance(value: Date | undefined) {
    this._dateNaissance = value;
  }

  public set type(value: string | undefined) {
    this._type = value;
  }
}
